skimage/filters/_blur_effect.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+from scipy.ndimage import convolve
+
+from ..color import rgb2gray
+from . import sobel_h, sobel_v
+from ..util import img_as_float
+
+
+__all__ = ['blur_effect']
+
+
+def blur_effect(image, h_size=11):
+    """
+    Compute a metric that indicates the strength of blur in an image (0 for no
+    blur, 1 for maximal blur).
+
+    Parameters
+    ----------
+    image : ndarray
+        RGB or grayscale 2D image. The input image is converted to grayscale
+        before computing the blur metric.
+    h_size : int, optional
+        Size of the re-blurring filter. Default is 11.
+
+    Returns
+    -------
+
+    blur : float (0 to 1)
+        Blur metric: the maximum of horizontal (Bx) and vertical (By) metrics.
+    blur_table : ndarray [Bx, By]
+        Bx (resp. By) is the blur metric in the horizontal (resp. vertical)
+        direction.
+
+    Notes
+    -----
+
+    `h_size` must keep the same value in order to compare results between
+    images. Most of the time, the default size (11) is enough. This means that
+    the metric can clearly discriminate blur up to an average 11x11 filter; if
+    blur is higher, the metric still gives good results but its values tend
+    towards an asymptote.
+
+    References
+    ----------
+    .. [1] Frederique Crete, Thierry Dolmiere, Patricia Ladret, and Marina
+       Nicolas "The blur effect: perception and estimation with a new
+       no-reference perceptual blur metric" Proc. SPIE 6492, Human Vision and
+       Electronic Imaging XII, 64920I (2007)
+       https://hal.archives-ouvertes.fr/hal-00232709
+       :DOI:`10.1117/12.702790`
+    """
+    if image.ndim not in (2, 3):
+        raise ValueError('image must be 2-dimensional')
+
+    if image.ndim == 3:
+        image = rgb2gray(image)
+    image = img_as_float(image)
+    shape = image.shape
+
+    h = np.ones((h_size, 1)) / h_size
+
+    # vertical blur
+    ver = convolve(image, h)
+    im_sharp = np.abs(sobel_h(image))
+    im_blur = np.abs(sobel_h(ver))
```

Comment:
I prefer `sobel` with axis=0 or axis=1 to `sobel_h` or `sobel_v`. I think that will be less confusing and is easier to see how we could refactor to support n-dimensions.

Reply:
:tada: nD support: 09c0379e1121619229 please review :wink: